certora/README.md

```diff
@@ -1,35 +1,223 @@
+
+Note also that these properties are ensured to always hold, because the contract is verified to be immutable in [`Immutability.spec`](specs/Immutability.spec).
+
+### Consistent asset
+
+For deposit and withdraw, it is checked that markets have a the same loan token as the loan token of the vault.
+We say that such market has a consistent asset.
+The following function defined in [`ConsistentState.spec`](specs/ConsistentState.spec) is verified to always return `true` and contributes to verifying the property above.
+
+```solidity
+function isEnabledHasConsistentAsset(MetaMorphoHarness.MarketParams marketParams) returns bool {
+    MetaMorphoHarness.Id id = Morpho.libId(marketParams);
+
+    return config_(id).enabled => marketParams.loanToken == asset();
+}
+```
+
+This checks that each market in the withdraw queue has a consistent asset.
 
-# Folder and file structure
+### Enabled flag
+
+It is verified that the enabled flag tracks if a given market is in the withdraw queue.
+The following function defined in [`Enabled.spec`](specs/Enabled.spec) gives one half of this property, and is verified to always return `true` for every index `i`.
+
+```solidity
+function isInWithdrawQueueIsEnabled(uint256 i) returns bool {
+    if(i >= withdrawQueueLength()) return true;
+
+    MetaMorphoHarness.Id id = withdrawQueue(i);
+
+    return config_(id).enabled;
+}
+```
+
+Thorough the code of MetaMorpho, this enabled flag is checked to be set to true, which is an efficient check that does not require to go through the whole withdraw queue.
+This check also ensures that such markets have some properties, since verifying those are necessary to be added to the withdraw queue.
+For example, markets added to the withdraw queue necessarily have a consistent asset.
+
+### Last updated
+
+The `lastUpdate` variable in Morpho Blue is used to discriminate if a market is created.
+In MetaMorpho, every market in the supply queue or in the withdraw queue has been created.
+
+```solidity
+rule newSupplyQueueEnsuresPositiveCap(env e, MetaMorphoHarness.Id[] newSupplyQueue) {
+    uint256 i;
+
+    setSupplyQueue(e, newSupplyQueue);
+
+    MetaMorphoHarness.Id id = supplyQueue(i);
+
+    assert config_(id).cap > 0;
+}
+```
+
+The previous rule ensures that when setting a new supply queue, each market has a positive supply cap.
+Markets can have a positive supply cap only if they are created on Morpho Blue.
+We can prove then that each market of the supply queue has been created on Morpho, because if a market has ben created it cannot be "destroyed" later.
+
+## Liveness
+
+The liveness properties ensures that some crucial actions cannot be blocked.
+It is notably useful to show that in case of an emergency, it is still possible to make salvaging transactions.
+The `canPauseSupply` rule in [`Liveness.spec`](specs/Liveness.spec) shows that it is always possible to pause supply.
```

Comment:
"pause" sounds weird
```suggestion
The `canPauseSupply` rule in [`Liveness.spec`](specs/Liveness.spec) shows that it is always possible to prevent new deposits.
```